libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -178,9 +179,18 @@ Json::Value ASTJsonConverter::inlineAssemblyIdentifierToJson(pair<yul::Identifie
 	tuple["declaration"] = idOrNull(_info.second.declaration);
 	tuple["isSlot"] = Json::Value(_info.second.suffix == "slot");
 	tuple["isOffset"] = Json::Value(_info.second.suffix == "offset");
+
 	if (!_info.second.suffix.empty())
+	{
 		tuple["suffix"] = Json::Value(_info.second.suffix);
+
+		static auto specialSuffixes = std::set{"slot", "offset", "selector", "address"};
```

Comment:
Wait, do we actually need this? What does `suffix` contain if we have `.slot` without this actually being the slot of a state variable? Or is it always the slot of a state variable? Can we have
```
contract C { uint constant slot = 1; }
contarct D {
function f() public pure { assembly { let x := C.slot } }
}
```

Reply:
If we don't need it, we don't need any extra `specialAccess` attribute either, don't we? Because if there is a suffix it is special access, all the checks are done beforehand anyway...